frontend/src/features/admin-form/settings/SettingsPage.tsx

```diff
@@ -50,6 +52,25 @@ export const SettingsPage = (): JSX.Element => {
   const displayPayments =
     user?.betaFlags?.payment || flags?.has(featureFlags.payment)
 
+  // Note: Admins are not redirected to /general on invalid settings tabs.
+  useEffect(() => {
+    if (displayPayments) {
+      settingsTabsOrder.push('payments')
+      setTabIndex(settingsTabsOrder.indexOf(settingsTab ?? ''))
+    }
+  }, [displayPayments, settingsTab])
+
+  const [tabIndex, setTabIndex] = useState(
```

Comment:
Not sure about the usage of setState here, I think can just const tabIndex, without the useEffect above, the setState isn't really needed?

Reply:
The `useEffect` was needed because we need to wait for displayPayments to load first then do the navigation

Reply:
Ahh.. Okay I get what you mean now! Thanks for the explanation!